static/js/components/NewDropDownSearchBar.jsx

```diff
@@ -0,0 +1,54 @@
+// NewDropDownSearchBar.jsx
+import React from "react";
+import PropTypes from "prop-types";
+import Autocomplete from "@mui/material/Autocomplete";
+import TextField from "@mui/material/TextField";
+
+const NewDropDownSearchBar = ({
+  optionsList,
+  selectedOption,
+  onOptionChange,
+  searchValue,
+  onSearchChange,
+  label,
+}) => (
+  <Autocomplete
+    options={optionsList}
+    getOptionLabel={(option) => `${option.telescopeName} / ${option.name}`}
```

Comment:
The component can't be generic if hardcoded to display this text specific to the telescopes. Also, I currently see this because of it:
<img width="463" alt="image" src="https://github.com/skyportal/skyportal/assets/49785117/93ed4ea2-1255-4a16-acbd-2194beb873b6">

because `telescopeName` doesn't not exist in the instrumentList you passed as a parameter. 

What you want, is to build the right option list before passing it to this component.

Here for instance, instead of passing the plain instrumentList, you can create an instrumentOptionList array, where each element has: an id, a label.

Does that make sense? You want the component to take a generic list of options with an id and a label you build before hand.

Here, to add the telescopeName, in the NewObservations.jsx you'll need to either get the telescopeList to and find the match for each by id, or check if the instrument contains a telescope object which does have the name. A good old console.log will do the trick :)

Reply:
I am not 100% sure but I am finding this failure on other pages such as Allocations page as shown below where I have not modified the original codebase. Do you think I am not successfully loading instrument lists inside my environment?

<img width="1840" alt="image" src="https://github.com/skyportal/skyportal/assets/105004472/81eaeaae-8921-4374-842a-b1edcd211e59">

Reply:
What do you mean by not loading successfully? In your frontend code, you call your newly added component and pass it a list of options. In your component, as I mentioned in my review, you are looking for a `telescopeName` that doesnt even exist. 

Locally, with the code on main, that does not happen at all. That is introduces by your changes.

As I mentioned in the last review, you want to not just pass `instrumentList` that you just fetched from the redux with `useSelector`, but you want to create a new list which element can have the instrument id, and name/label. But you need to build that label there and not hardcode that you need instrument name in the component you created, which effectively does not make it generic at all if you build the name in it.

and to get the telescopeName, well as it is already done in many places of the code you get the list of telescopes from the redux with `useSelector` and you can build a simple mapper from the telescope id to its name, so you can easily figure out what's the name of the telescope associated to an instrument.

Doest that make sense?

Reply:
However, I do not understand why you would see that on existing pages where you do not use your component. There could be another problem on top of the one I mentioned in my latest review. Please check in the console if the list of telescopes in the redux is there or missing.